lib/sparse/src/index/posting_list.rs

```diff
@@ -0,0 +1,179 @@
+use crate::common::types::{DimWeight, RecordId};
+
+#[derive(Debug, Copy, Clone)]
+pub struct PostingElement {
+    pub id: RecordId,
+    pub weight: DimWeight,
+    pub max_next_weight: DimWeight,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct PostingList {
+    /// List of the posting elements ordered by id
+    pub elements: Vec<PostingElement>,
+}
+
+impl PostingList {
+    #[cfg(test)]
+    pub fn from(records: Vec<(RecordId, DimWeight)>) -> PostingList {
+        let mut posting_list = PostingBuilder::new();
+        for (id, weight) in records {
+            posting_list.add(id, weight);
+        }
+        posting_list.build()
+    }
+}
+
+pub struct PostingBuilder {
+    elements: Vec<PostingElement>,
+}
+
+impl PostingBuilder {
+    pub fn new() -> PostingBuilder {
+        PostingBuilder {
+            elements: Vec::new(),
+        }
+    }
+
+    pub fn add(&mut self, id: RecordId, weight: DimWeight) {
+        self.elements.push(PostingElement {
+            id,
+            weight,
+            max_next_weight: f32::NEG_INFINITY,
+        });
+    }
+
+    pub fn build(mut self) -> PostingList {
+        // Sort by id
+        self.elements.sort_by_key(|e| e.id);
```

Comment:
[`sort_unstable_by_key`](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable_by_key)?

```suggestion
        self.elements.sort_unstable_by_key(|e| e.id);
```